### ISSUE TITLE
Notification.verify() needs to take a http_options (i.e. client id/secret config) arg as per the spec of executeHttp()

### ISSUE THREAD

**Issue by dcworldwide** [paypal/PayPal-node-SDK#426]
I have a multi-tenant system. There is no single REST api client/secret that I can use to verify webhook events...I need to use the tenants config.

Notification.verify() does not allow me to pass one in. verify() calls executeHttp() which accepts a http_options arg where you can set the client id/secret....verify() signature needs to change so I can pass these values down.

**Comment by dcworldwide** [comment]
Here is a monkey patch solution as an example only

```
function verify2(headers, body, webhookId, conifg, callback) {
        // In an effort not to break existing applications, accept old arguments temporarily
        if (arguments.length > 4) {
            /* jshint validthis: true */
            return verifyLegacy.apply(this, arguments);
        }

        if (typeof headers !== 'object') {
            return callback(new Error("headers is not an object"), false);
        }

        // Normalizes headers
        Object.keys(headers).forEach(function (header) {
            headers[header.toUpperCase()] = headers[header];
        });

        var webhookEventBody = (typeof body === "string") ? JSON.parse(body) : body;

        var payload = {
            'auth_algo': headers['PAYPAL-AUTH-ALGO'],
            'cert_url': headers['PAYPAL-CERT-URL'],
            'transmission_id': headers['PAYPAL-TRANSMISSION-ID'],
            'transmission_sig': headers['PAYPAL-TRANSMISSION-SIG'],
            'transmission_time': headers['PAYPAL-TRANSMISSION-TIME'],
            'webhook_id': webhookId,
            'webhook_event': webhookEventBody
        };

        api.executeHttp('POST', '/v1/notifications/verify-webhook-signature', payload, conifg, callback);
    }
```

**Comment by sucithrashanu** [comment]
@dcworldwide 
You are about to ask multiple client/secret (array of objects) in config to execute in the executeHttp()?

**Comment by sucithrashanu** [comment]
@dcworldwide 
Refer This might be helpful if your about to use Muti-client/secret in config:
https://github.com/paypal/PayPal-node-SDK/tree/master/samples/configuration

**Comment by dcworldwide** [comment]
Hi, I think my bug description and proposed solution was clear. That link is related but doesn't address the issue.

**Comment by sucithrashanu** [comment]
@dcworldwide 

Did your [monkey patch solution](https://github.com/paypal/PayPal-node-SDK/issues/426#issuecomment-502002188) resolved your problem?

Thanks

**Comment by dcworldwide** [comment]
Yes it did.... I would like a PayPal developer takes the patch and applies it to the library

**Comment by johnsonsan** [comment]
Unsure what to be done in this case. Nothing

**Comment by prakash-gangadharan** [comment]
Hi @dcworldwide,
* If you find solution to an [issue/improvements](https://github.com/paypal/PayPal-node-SDK/issues) in sdk that would be helpful to everyone, feel free to send us a pull request.
* The ideal approach to create a fix would be to fork the repository, create a branch in your repository, and make a pull request out of it.
* It is desirable if there is enough comments/documentation and Tests included in the pull request.
* For general idea of contribution, please follow the guidelines mentioned [here](https://guides.github.com/activities/contributing-to-open-source/).

**Comment by dcworldwide** [comment]
@prakash-gangadharan I don't have time. But, the solution is above for someone who `works` for PP to fix.

**Comment by gastonelhordoy** [comment]
Hey @prakash-gangadharan, I think the main issue here is that this repo is actually part of the **integration SDK provided by PayPal**, a company that actually supports the development of these tools so developers can more easily integrate with it and therefore make PayPal earn more money.

So this is not Open Source strictly speaking like in the case of a developer creating a library on her/his free time and depending on other's contributions. PayPal is paying a salary to you and other developers so you improve these tools.

Furthermore, @dcworldwide has even provided the code to fix, it's just a matter for you to apply that change and that's it!

We would really appreciate if you could do your job, thanks!
